### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by camhashemi**
@gratimax @ryanpbrewster not sure if this is even more readable, albeit more scala-esque

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/OrganizationDomainOwnershipRepo.scala [hunk 1]
```diff
@@ -54,24 +54,16 @@ class OrganizationDomainOwnershipRepoImpl @Inject() (
...
 
-  override def getDomainOwnershipBetween(organization: Id[Organization], domainHostname: NormalizedHostname, onlyState: Option[State[OrganizationDomainOwnership]] = Option(OrganizationDomainOwnershipStates.ACTIVE))(implicit session: RSession): Option[OrganizationDomainOwnership] = {
-    val query =
-      if (onlyState.isDefined) {
-        for { row <- rows if row.organizationId === organization && row.domainHostname === domainHostname && row.state === onlyState } yield row
-      } else {
-        for { row <- rows if row.organizationId === organization && row.domainHostname === domainHostname } yield row
-      }
-    query.firstOption
+  override def getDomainOwnershipBetween(organization: Id[Organization], domainHostname: NormalizedHostname, excludeState: Option[State[OrganizationDomainOwnership]] = Some(OrganizationDomainOwnershipStates.INACTIVE))(implicit session: RSession): Option[OrganizationDomainOwnership] = {
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/model/OrganizationDomainOwnershipRepo.scala:57]: Does this need to be an override? There's no base implementation, right?

Comment (maxov) [inline kifi-backend/modules/shoebox/app/com/keepit/model/OrganizationDomainOwnershipRepo.scala:57]: You can override abstract methods defined in parent traits. This means that if the method in the parent trait is removed or the signature is changed, a compiler error results. I think that this is generally good practice, but others may have opinions on it.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationDomainOwnershipCommander.scala [hunk 2]
```diff
@@ -71,25 +71,17 @@ class OrganizationDomainOwnershipCommanderImpl @Inject() (
...
-              }
-          }
+      NormalizedHostname.fromHostname(domainName).fold[Either[OwnDomainFailure, OwnDomainSuccess]](ifEmpty = Left(InvalidDomainName(domainName))) { normalizedHostname =>
+        domainRepo.get(normalizedHostname).fold[Either[OwnDomainFailure, OwnDomainSuccess]](ifEmpty = Left(DomainDidNotExist(domainName))) { domain =>
+          orgDomainOwnershipRepo.getOwnershipForDomain(domain.hostname, excludeState = None).fold[Either[OwnDomainFailure, OwnDomainSuccess]](
+            ifEmpty = Right(OwnDomainSuccess(domain, orgDomainOwnershipRepo.save(orgDomainOwnershipRepo.save(OrganizationDomainOwnership(organizationId = orgId, normalizedHostname = domain.hostname)))))
+          ) {
+              case ownership if ownership.state == OrganizationDomainOwnershipStates.INACTIVE =>
+                Right(OwnDomainSuccess(domain, orgDomainOwnershipRepo.save(ownership.copy(state = OrganizationDomainOwnershipStates.ACTIVE))))
+              case ownership if ownership.organizationId != orgId => Right(OwnDomainSuccess(domain, ownership))
```
Comment (maxov) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationDomainOwnershipCommander.scala]: I don't think we mean to return a success here, if the domain is owned by somebody else it means that we cannot claim it.

**Comment by maxov** [comment]
:+1: I think it's more readable, big improvement. Nice!

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationDomainOwnershipCommander.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/OrganizationDomainOwnershipCommander.scala]: nice catch :8ball:

**Pull request merged**